sabre1041/managing-ocp-install-beyond
# Refactor adding features to streamline lab

This PR adds several features to make deployment more efficient and streamlined. Several issues are addressed as detailed in the next section.

## Changes
- Remove EBS volumes on terminate playbook created and bound to any OCP infra or apps. 
- Add `aws_lab_terminate.yml` playbook to cleanup all environments specified in an inventory file (`tower_instances`)
- Remove need to specify `student_id` in the `tower_instances` inventory file, as this will be determined form the inventory_hostname
- Combine `tower_config` and `tower_config_type` vars
- Change `tag:Name` to be fqdn instead of previous value of `rhte-<role>-<student_id>`
- Remove need for specifying AWS access/secret key in playbooks run from Tower instead sourcing from environment
- Move source vars for AWS inventory group task to default variables
- Add wait: yes to all route53 tasks to wait for DNS propagation
- Change name of hosts to fqdn `tag:Name` but specify `ansible_host` to be `ec2_public_ip` to avoid any issues with DNS resolution
- Add `wait_for` to use `host` command on Tower to wait for resolution of fqdn before moving on
- Change terminate job launch to be fully asynchronous and non-blocking

## Instructions
- Remove any mention of `tower_config` in your secrets file
- Rename any variables in secrets files from `tower_config_type` to `tower_config`. There is no longer a need to specify `tower_config: true` instead provide the values `tower_config: none|self|full|test`
- Pull up AWS console for ec2 instances, ebs volumes and route53 in 3 different tabs and remove any entries matching your `lab_user` or `student_id`

## Test 1
Deploy environments with these variables:
```
student_count: 3
tower_config: test
tower_project_provision_and_configure_url: "https://github.com/vvaldez/summit-2017-ocp-operator.git"
tower_project_provision_and_configure_branch: "cleanup-volumes-on-terminate"
```

- Verify OCP cluster works and optionally deploy a couple of quick cake+mysql apps on the first cluster and verify the EBS volumes show up on AWS
- Execute the `Terminate_OCP` job template and verify only OCP cluster was deleted and Tower remains
- Verify the EBS volumes related to this cluster are deleted from AWS
- If successful then execute the workflow template `3-Terminate_All` which will terminate tower.
- Verify all instances, volumes and route53 entries are removed

## Test 2:
There is also an `aws_lab_terminate.yml` which is designed to run against whatever specified tower instances in the inventory file and launch the terminate workflow. This means that an entire lab of tower and fully deployed OCP clusters created can be deleted at once. So, if you deployed 3 clusters the first one should be gone. Now create an inventory file called `tower_instances` with this content:
```
[tower_instances]
tower-vvaldez-[2:3].rhte.sysdeseng.com

[all:vars]
ansible_ssh_user=ec2-user
```

Now execute the new playbook as such:
```
ansible-playbook -i tower_instances aws_lab_terminate.yml -e my_secrets.yml
```

Watch the job progress and after a few minutes verify in AWS that the EBS volumes have been removed along with the remaining clusters, and all instances, volumes and route53 entries are removed.
